api/authentication/models.py

```diff
@@ -131,23 +165,106 @@ def get_short_name(self):
         short_name = "%s" % self.first_name
         return short_name.strip()
 
-    def email_user(self, subject, message, from_email=None, **kwargs):
-        """Send an email to this user."""
-        send_mail(subject, message, from_email, [self.email], **kwargs)
-
     def get_avatar(self):
         """ Returns avatar (use Pillow) """
         pass
 
+    def get_verification_code(self,email):
+        # verification token 
+        signer = TimestampSigner()
+        return b64_encode(bytes(signer.sign(email), encoding='utf-8'))
+        
+    @classmethod
+    def verify_email(self,code):
+        if code:
+            signer = TimestampSigner()
+            try:
+                code = code.encode('utf-8')
+                max_age = datetime.timedelta(days=base.VERIFICATION_CODE_EXPIRED).total_seconds()
+                code = force_bytes(code)
+                code = b64_decode(code)
+                code = code.decode()
+                email = signer.unsign(code, max_age=max_age)
+                user = StdUser.objects.get(**{StdUser.USERNAME_FIELD:email,'is_active':False})
+                print('do delete ::::',user.code)
+                print(user.email)
+                if user.is_active == True:
+                    return False
+                user.is_active = True
+                user.code = "None code"
+                user.save()
+                return True, ('Your account has been activated.')  
+            except SignatureExpired:
+                return False, ('Your code to activate by link expired')
+            except (BadSignature, StdUser.DoesNotExist, TypeError, UnicodeDecodeError) as e:
+                print(e)
+                pass
+            return False, ('Activation link is incorrect, please resend request')
+    
+    @classmethod
+    def verify_password(self,code,password):
+        if code:
```

Comment:
the same questions
What will be if no code in args? please update logic for this method
What happens when this if statement fails?